packages/sui-react-router/src/Router.js

```diff
@@ -66,15 +66,26 @@ const Router = ({
 
   useEffect(() => {
     let prevState = {}
+    let isSkipped = !!matchContext
```

Comment:
If `matchContext` is provided that means that we can avoid the `setState` being execute since the router properties are going to be the same